pkg/glib/glib.go

```diff
@@ -131,9 +244,121 @@ func createGSuiteUserFromConfig(user *config.UserConfig) *admin.User {
 		googleUser.Emails = workEm
 	}
 
+	if user.Employee != (config.EmployeeConfig{}) {
+		uOrg := []admin.UserOrganization{
+			{
+				Department:  user.Employee.Department,
+				Title:       user.Employee.JobTitle,
+				CostCenter:  user.Employee.CostCenter,
+				Description: user.Employee.Type,
+			},
+		}
+		googleUser.Organizations = uOrg
+		rel := []admin.UserRelation{
+			{
+				Value: user.Employee.ManagerEmail,
+				Type:  "manager",
+			},
+		}
+		googleUser.Relations = rel
+
+		ids := []admin.UserExternalId{
+			{
+				Value: user.Employee.EmployeeID,
+				Type:  "organization",
+			},
+		}
+		googleUser.ExternalIds = ids
+
+	}
+
+	if user.Location != (config.LocationConfig{}) {
+		loc := []admin.UserLocation{
+			{
+				Area:         "desk",
+				BuildingId:   user.Location.Building,
+				FloorName:    user.Location.Floor,
+				FloorSection: user.Location.FloorSection,
+				Type:         "desk",
+			},
+		}
+		googleUser.Locations = loc
+
+	}
+
 	return googleUser
 }
 
+// createConfigUserFromGSuite converts a (GSuite) admin.User to ConfigUser
+func CreateConfigUserFromGSuite(googleUser *admin.User) config.UserConfig {
+	// get emails
+	primaryEmail, secondaryEmail := GetUserEmails(googleUser)
+
+	configUser := config.UserConfig{
+		FirstName:      googleUser.Name.GivenName,
+		LastName:       googleUser.Name.FamilyName,
+		PrimaryEmail:   primaryEmail,
+		SecondaryEmail: secondaryEmail,
+		OrgUnitPath:    googleUser.OrgUnitPath,
+		RecoveryPhone:  googleUser.RecoveryPhone,
+		RecoveryEmail:  googleUser.RecoveryEmail,
+	}
+
+	if len(googleUser.Aliases) > 0 {
+		for _, alias := range googleUser.Aliases {
+			configUser.Aliases = append(configUser.Aliases, string(alias))
+		}
+	}
+
+	if googleUser.Phones != nil {
+		for _, phone := range googleUser.Phones.([]interface{}) {
+			configUser.Phones = append(configUser.Phones, fmt.Sprint(phone.(map[string]interface{})["value"]))
+		}
+	}
+
+	if googleUser.ExternalIds != nil {
+		for _, id := range googleUser.ExternalIds.([]interface{}) {
+			if id.(map[string]interface{})["type"] == "organization" {
+				configUser.Employee.EmployeeID = fmt.Sprint(id.(map[string]interface{})["value"])
+			}
+		}
+	}
+
+	if googleUser.Organizations != nil {
+		for _, org := range googleUser.Organizations.([]interface{}) {
+			configUser.Employee.Department = fmt.Sprint(org.(map[string]interface{})["department"])
+			configUser.Employee.JobTitle = fmt.Sprint(org.(map[string]interface{})["title"])
+			configUser.Employee.Type = fmt.Sprint(org.(map[string]interface{})["description"])
+			configUser.Employee.CostCenter = fmt.Sprint(org.(map[string]interface{})["costCenter"])
+		}
```

Comment:
In the future we maybe should find a way to write this in a saner way. These many inline type assertions are a big code smell and I cannot understand why the GSuite SDK thingy is so unspecific/unhelpful.

Nothing that needs to be addressed in this PR though.